packages/twenty-front/src/modules/apollo/optimistic-effect/utils/triggerCreateRecordsOptimisticEffect.ts

```diff
@@ -0,0 +1,112 @@
+import { ApolloCache, StoreObject } from '@apollo/client';
+
+import { isCachedObjectConnection } from '@/apollo/optimistic-effect/utils/isCachedObjectConnection';
+import { CachedObjectRecord } from '@/apollo/types/CachedObjectRecord';
+import { CachedObjectRecordEdge } from '@/apollo/types/CachedObjectRecordEdge';
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+import { capitalize } from '~/utils/string/capitalize';
+
+/*
+  TODO: for now new records are added to all cached record lists, no matter what the variables (filters, orderBy, etc.) are.
+  We need to refactor how the record creation works in the RecordTable so the created record row is temporarily displayed with a local state,
+  then we'll be able to uncomment the code below so the cached lists are updated coherently with the variables.
+*/
+export const triggerCreateRecordsOptimisticEffect = ({
+  cache,
+  objectMetadataItem,
+  records,
+}: {
+  cache: ApolloCache<unknown>;
+  objectMetadataItem: ObjectMetadataItem;
+  records: CachedObjectRecord[];
+}) => {
+  const objectEdgeTypeName = `${capitalize(
+    objectMetadataItem.nameSingular,
+  )}Edge`;
+
+  cache.modify<StoreObject>({
+    fields: {
+      [objectMetadataItem.namePlural]: (
+        cachedConnection,
+        {
+          INVALIDATE: _INVALIDATE,
+          readField,
+          storeFieldName: _storeFieldName,
+          toReference,
+        },
+      ) => {
+        if (
+          !isCachedObjectConnection(
+            objectMetadataItem.nameSingular,
+            cachedConnection,
+          )
+        )
+          return cachedConnection;
+
+        /* const { variables } =
```

Comment:
I'd rather keep it if that's ok, as I'll work on it this week! For the requirements, they're pretty straightforward. In order for the cached list to be coherent, the record should only be added if it matches the list's `filters`. Then if the record is added to the list, it needs to be re-ordered according to the `orderBy` variable and then the array of edges needs to be trimmed to the correct length if the list has a `first` (= `limit`) variable. But this will only be possible when we implement the "added record" as a local state in the `RecordTable`, which I'll work on soon.